firefox-ios/firefox-ios-tests/Tests/ClientTests/TabTray/InactiveTabsTelemetryTests.swift

```diff
@@ -9,49 +9,73 @@ import XCTest
     }
 
     override func tearDown() {
         subject = nil
+        gleanWrapper = nil
         super.tearDown()
     }
 
     func testRecordInactiveTabWhenSectionShownThenGleanIsCalled() throws {
         subject?.sectionShown()
 
-        testCounterMetricRecordingSuccess(metric: GleanMetrics.InactiveTabsTray.inactiveTabShown)
+        XCTAssertEqual(gleanWrapper.submitCounterMetricTypeCalled, 1)
+        if let savedMetric = gleanWrapper.savedEvent as? CounterMetricType {
+            XCTAssert(type(of: savedMetric) == type(of: GleanMetrics.InactiveTabsTray.inactiveTabShown))
+        } else {
+            XCTFail("savedEvent is not of type CounterMetricType")
+        }
     }
 
     func testRecordInactiveTabWhenClosedAllTabsThenGleanIsCalled() throws {
         subject?.closedAllTabs()
 
-        testCounterMetricRecordingSuccess(metric: GleanMetrics.InactiveTabsTray.inactiveTabsCloseAllBtn)
+        XCTAssertEqual(gleanWrapper.submitCounterMetricTypeCalled, 1)
+        if let savedMetric = gleanWrapper.savedEvent as? CounterMetricType {
+            XCTAssert(type(of: savedMetric) == type(of: GleanMetrics.InactiveTabsTray.inactiveTabsCloseAllBtn))
+        } else {
+            XCTFail("savedEvent is not of type CounterMetricType")
+        }
     }
 
     func testRecordInactiveTabWhenTabOpenedThenGleanIsCalled() throws {
         subject?.tabOpened()
 
-        testCounterMetricRecordingSuccess(metric: GleanMetrics.InactiveTabsTray.openInactiveTab)
+        XCTAssertEqual(gleanWrapper.submitCounterMetricTypeCalled, 1)
+        if let savedMetric = gleanWrapper.savedEvent as? CounterMetricType {
+            XCTAssert(type(of: savedMetric) == type(of: GleanMetrics.InactiveTabsTray.openInactiveTab))
+        } else {
+            XCTFail("savedEvent is not of type CounterMetricType")
+        }
     }
 
     func testRecordInactiveTabWhenTabSwipedClosedThenGleanIsCalled() throws {
         subject?.tabSwipedToClose()
 
-        testCounterMetricRecordingSuccess(metric: GleanMetrics.InactiveTabsTray.inactiveTabSwipeClose)
+        XCTAssertEqual(gleanWrapper.submitCounterMetricTypeCalled, 1)
+        if let savedMetric = gleanWrapper.savedEvent as? CounterMetricType {
+            XCTAssert(type(of: savedMetric) == type(of: GleanMetrics.InactiveTabsTray.inactiveTabSwipeClose))
+        } else {
+            XCTFail("savedEvent is not of type CounterMetricType")
+        }
     }
 
     func testRecordInactiveTabWhenThenGleanIsCalled() throws {
         subject?.sectionToggled(hasExpanded: true)
 
-        testEventMetricRecordingSuccess(metric: GleanMetrics.InactiveTabsTray.toggleInactiveTabTray)
+        XCTAssertEqual(gleanWrapper.submitEventMetricTypeCalled, 1)
+        if let savedMetric
+            = gleanWrapper.savedEvent as? EventMetricType<GleanMetrics.InactiveTabsTray.ToggleInactiveTabTrayExtra> {
+            XCTAssert(type(of: savedMetric) == type(of: GleanMetrics.InactiveTabsTray.inactiveTabSwipeClose))
```

Comment:
failing locally, here's the event that passes it
```suggestion
            XCTAssert(type(of: savedMetric) == type(of: GleanMetrics.InactiveTabsTray.toggleInactiveTabTray))
```

Reply:
```
XCTAssert(type(of: savedMetric) == type(of: GleanMetrics.InactiveTabsTray.toggleInactiveTabTray))
```
Also, for this part, when it fails, I don't get a valuable test failure message. Unfortunately, the event type is not equatable so I couldn't find a better way to use the built in assertions. To combat this, what about adding a message? Off the top of my head, maybe something like this?

```
let debugMessage = "Expected savedMetric to be of type \(type(of: GleanMetrics.InactiveTabsTray.toggleInactiveTabTray)), but got \(type(of: savedMetric))"
XCTAssert(type(of: savedMetric) == type(of: GleanMetrics.InactiveTabsTray.toggleInactiveTabTray))
```